src/status_im/contexts/wallet/send/routes/view.cljs

```diff
@@ -202,18 +202,20 @@
       :on-press-to-network   on-press-to-network}]))
 
 (defn fetch-routes
-  [amount routes-can-be-fetched? bounce-duration-ms]
-  (if routes-can-be-fetched?
+  [amount valid-input? bounce-duration-ms]
```

Comment:
This function was the cause of the problem. It was getting too generic condition `routes-can-be-fetched?` that was made from two logical conditions - if the valid is input and if we are still on the routes screen.
And when user navigated to confirmation page that condition became false which lead to clearing suggested routes.

Now in this function we rely only on input validity. And when the screen is wrong we dont't even call it.